samples/snippets/conftest.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import uuid
+
+import google.auth
+from google.cloud import bigquery
+from google.cloud import datacatalog_v1
+
+import pytest
+
+bigquery_client = bigquery.Client()
+
+
+@pytest.fixture(scope="session")
+def client(credentials):
+    return datacatalog_v1.DataCatalogClient(credentials=credentials)
+
+
+@pytest.fixture(scope="session")
+def default_credentials():
+    return google.auth.default()
+
+
+@pytest.fixture(scope="session")
+def credentials(default_credentials):
+    return default_credentials[0]
+
+
+@pytest.fixture(scope="session")
+def project_id(default_credentials):
+    return default_credentials[1]
+
+
+@pytest.fixture
+def dataset_id():
+    now = datetime.datetime.now()
+    dataset_id = "python_dataset_sample_{}_{}".format(
+        now.strftime("%Y%m%d%H%M%S"), uuid.uuid4().hex[:8]
```

Comment:
This logic is used repeatedly. Let's define a `temp_suffix()` function to make these fixtures a little less verbose.

I did something similar here: https://github.com/googleapis/python-bigquery/pull/408/files#diff-4833cde58155bb800c7c77901b7044ec5d9b40f1224290a0d429fd547328bcc2R25 but I like your use of the current day for easier debugging